lib/deimos/active_record_consume/batch_consumption.rb

```diff
@@ -170,15 +170,27 @@ def upsert_records(messages)
         updater.mass_update(record_list)
       end
 
+      # Returns a lookup entity to be used during record attributes
+      # @param _messages [Array<Deimos::Message>]
+      # @return [Hash, Set, ActiveRecord::Relation]
+      def record_lookup(_messages)
```

Comment:
I don't really understand what this is for. Can we explain this in the README?

I'd rather not make changes to the `record_attributes` method if possible. I'm not sure what this is, but can it be put in metadata somehow? Or maybe give `Deimos::Message` a `metadata` field for custom data and have a pre-process method to populate it?

Reply:
In item platform, before consuming local offers, we preload and create a lookup relation where we map merchant_id and merchant_store_code to a store id. 
```

store_set = messages.map { |m|
        { merchant_id: m.payload.merchant_id, merchant_store_code: m.payload.store_code }
      }.to_set
stores= store_set.reduce(Store.none) { |store, h| store.or(Store.where(h)) }.
        select(:id, :merchant_store_code, :merchant_id)
```
In record_attributes
```
id = stores.
        find { |store|
          store.merchant_store_code == store_code &&
            store.merchant_id == merchant_id
        }&.id
```

I think it makes sense to do this lookup and assignment in a pre_process method and attach the result to the deimos message as you suggested.
I'll look into implementing that